addons/spreadsheet_account/models/account.py

```diff
@@ -13,6 +13,87 @@
 class AccountMove(models.Model):
     _inherit = "account.account"
 
+    #--------------------------#
+    # Domain utility functions #
+    #--------------------------#
+
+    def _build_date_domain(self, date_period, company_id, include_initial_balance_filter):
+        ''' Returns the date domain for the given period and company. The returned domain
+            is different depending on whether or not we include the initial balance filter.
+        '''
+
+        company = self.env["res.company"].browse(company_id)
+        start, end = self._get_date_period_boundaries(
+            date_period, company
+        )
+
+        if not include_initial_balance_filter:
```

Comment:
If you keep the account types hardcoded on payable/receivable, we never include the initial balance for those.
So that parameter seems useless

Reply:
Mmmh, maybe I'm not looking at the right think but the demo accounts seem to behave otherwise.
![image](https://github.com/odoo/odoo/assets/125546433/8e700d92-f24d-4c42-90d4-c4cf70aa5520)

Reply:
I meant "always", sorry

And that is what we should want

Reply:
Point granted!